WordPress/gutenberg
# Border Controls: Update control components to allow 40px height

Related:

- https://github.com/WordPress/gutenberg/issues/40893#issuecomment-1160061349

## What?

Adds the ability for the `BorderControl`, `BorderBoxControl`, and `BorderRadiusControl` components to be displayed with a `40px` height.

At this stage, this PR does not toggle the display of border controls to `40px` by default. If we wish to make these controls that height straight away we can close this PR in favor of more straightforward tweaks.

## Why?

The current design direction for new tools / controls is to have a `40px` default height.

<img src="https://user-images.githubusercontent.com/548849/174545138-bacefd5a-3095-4165-93ac-a92cb445ff71.png" width="400"/>

## How?

- Replaces the `__next36pxDefaultSize` prop for the `BorderControl` and `BorderBoxControl` components with `__next40pxDefaultSize`
- Updates the `BorderRadiusControl` to accept a new `__next40pxDefaultSize` prop
- Uses the `__next40pxDefaultSize` props to make each component display with a `40px` height
- The `BorderRadiusControl` leverages the `size` prop of its inner `UnitControl`/`InputControl` to make its input `40px`. 

## Testing Instructions

1. After checking out this PR, edit a post, add a Group block, and select it
2. Within the inspector controls sidebar the border controls should still be displayed within a 30px height
3. Switch to the site editor and within the Global Styles panel navigate to Blocks > Group > Layout
4. Confirm the border controls are still 30px
5. Apply the patch below to toggle on the 40px height for border controls
6. Ensure the border controls in both editors are now 40px high and all inner components are positioned correctly

<details>
<summary><strong>Diff to toggle on 40px height</strong></summary>

```diff
diff --git a/packages/block-editor/src/hooks/border-radius.js b/packages/block-editor/src/hooks/border-radius.js
index fa61bafa92..f062373a19 100644
--- a/packages/block-editor/src/hooks/border-radius.js
+++ b/packages/block-editor/src/hooks/border-radius.js
@@ -38,6 +38,7 @@ export function BorderRadiusEdit( props ) {
 		<BorderRadiusControl
 			values={ style?.border?.radius }
 			onChange={ onChange }
+			__next40pxDefaultSize={ true }
 		/>
 	);
 }
diff --git a/packages/block-editor/src/hooks/border.js b/packages/block-editor/src/hooks/border.js
index 81cc023bab..3766b56912 100644
--- a/packages/block-editor/src/hooks/border.js
+++ b/packages/block-editor/src/hooks/border.js
@@ -275,6 +275,7 @@ export function BorderPanel( props ) {
 						value={ hydratedBorder }
 						__experimentalHasMultipleOrigins={ true }
 						__experimentalIsRenderedInSidebar={ true }
+						__next40pxDefaultSize={ true }
 					/>
 				</ToolsPanelItem>
 			) }
diff --git a/packages/edit-site/src/components/global-styles/border-panel.js b/packages/edit-site/src/components/global-styles/border-panel.js
index 71fe697697..f9fcf1951f 100644
--- a/packages/edit-site/src/components/global-styles/border-panel.js
+++ b/packages/edit-site/src/components/global-styles/border-panel.js
@@ -186,6 +186,7 @@ export default function BorderPanel( { name } ) {
 						popoverOffset={ 40 }
 						__experimentalHasMultipleOrigins={ true }
 						__experimentalIsRenderedInSidebar={ true }
+						__next40pxDefaultSize={ true }
 					/>
 				</ToolsPanelItem>
 			) }
@@ -201,6 +202,7 @@ export default function BorderPanel( { name } ) {
 						onChange={ ( value ) => {
 							setBorderRadius( value || undefined );
 						} }
+						__next40pxDefaultSize={ true }
 					/>
 				</ToolsPanelItem>
 			) }

```
</details>

## Screenshots or screencast 

<details>
<summary><strong>Global Styles</strong></summary>

| Before | After |
|---|---|
| <img width="318" alt="Screen Shot 2022-06-22 at 5 34 22 pm" src="https://user-images.githubusercontent.com/60436221/174979605-38edff93-3439-4b2d-853a-049ba84f2d0a.png"> | <img width="302" alt="Screen Shot 2022-06-22 at 5 30 27 pm" src="https://user-images.githubusercontent.com/60436221/174979676-94d352d9-4a4d-4e9e-875a-c136a4aeda0c.png"> |
</details>


<details>
<summary><strong>Block Editor</strong></summary>

| Before | After |
|---|---|
| <img width="297" alt="Screen Shot 2022-06-22 at 6 18 19 pm" src="https://user-images.githubusercontent.com/60436221/174979907-d00e5b03-0c3f-4866-8e0a-9fd27bbb312d.png"> |  <img width="305" alt="Screen Shot 2022-06-22 at 4 52 15 pm" src="https://user-images.githubusercontent.com/60436221/174979997-e426469d-b073-472a-86c2-4c96bdc351a7.png"> |
</details>
